helper_test.go

```diff
@@ -88,6 +89,85 @@ func fetchTestAccountDetails(t *testing.T, client *Client) *TestAccountDetails {
 	return _testAccountDetails
 }
 
+func createAgent(t *testing.T, client *Client, org *Organization, agentPool *AgentPool, agentPoolToken *AgentToken) (*Agent, *AgentPool, func()) {
+	var orgCleanup func()
+	var agentPoolCleanup func()
+	var agentPoolTokenCleanup func()
+	var agent *Agent
+
+	if org == nil {
+		org, orgCleanup = createOrganization(t, client)
+	}
+
+	upgradeOrganizationSubscription(t, client, org)
+
+	if agentPool == nil {
+		agentPool, agentPoolCleanup = createAgentPool(t, client, org)
+		t.Log("create, log agentPool: ", agentPool)
+	}
+
+	if agentPoolToken == nil {
+		agentPoolToken, agentPoolTokenCleanup = createAgentToken(t, client, agentPool)
+	}
+
+	ctx := context.Background()
+	cmd := exec.Command("docker",
+		"run", "-d",
+		"--env", "TFC_AGENT_TOKEN="+agentPoolToken.Token,
+		"--env", "TFC_AGENT_NAME="+"this-is-a-test-agent",
+		"--env", "TFC_ADDRESS="+DefaultConfig().Address,
+		"docker.mirror.hashicorp.services/hashicorp/tfc-agent:latest")
+
+	go func() {
```

Comment:
This implementation was recommended to me, as feedback. What do you recommend in place of a goroutine? @nfagerlund

Reply:
If you're referring to me I actually [recommended the opposite](https://hashicorp.slack.com/archives/C01BR9FR8KZ/p1658428391943639?thread_ts=1657747733.956509&cid=C01BR9FR8KZ) 😅 

> One more thing to pile on -- since you're specifying the -d (detach) property in your docker run command you don't need to spin up a goroutine on the cmd.CombinedOutput. That command will finish as soon as the container has been created.

Reply:
Lol no, not you @taiidani. Thanks for linking your note. Working through lots of feedback.